### DISCUSSION TITLE
Physical models for real gases (friction and compressibility)

### DISCUSSION THREAD

**Discussion by dlohmeier** [e2nIEE/pandapipes#362] [Ideas]
I am currently trying to calculate a use case with real data from STANET, where the friction model "Prandtl-Colebrook 2" (whatever the difference between 1 and 2 exactly is) and the compressibility mode "GERG-88" were used. In general, STANET offers the following modes:

- friction models:
    - Prandtl-Colebrook
    - Interpolation (GANESI)
    - Nikuradse (smooth)
    - Nikuradse (continuous)
    - Prandtl-Colebrook 2
- compressibility models:
    - linear pressure dependency
    - AGA NX 19-mod
    - GERG-88

What we implemented in pandapipes by now are Prandtl-Colebrook (1 or 2?), Nikuradse smooth and Swamee-Jaine along with the linear pressure dependency. Can anyone give insights into the foundations of the other models, where to find literature on them, how they influence the results?

I think it would be great to be able to implement some additional models in pandapipes.

**Comment by jkisse** [comment]
There is a paper on GERG 2008/2004 here:
Kunz, O.; Wagner, W. (2012): The GERG-2008 Wide-Range Equation of State for Natural Gases and Other Mixtures: An Expansion of GERG-2004. In: J. Chem. Eng. Data 57 (11), S. 3032–3091. [DOI: 10.1021/je300655b](https://pubs.acs.org/doi/10.1021/je300655b).

In this paper, GERG (2008) and AGA-DC92 are compared. They also give a nice overview of different equations of state:
Clees, Tanja; Baldin, Anton; Klaassen, Bernhard; Nikitina, Lialia; Nikitin, Igor; Spelten, Philipp (2021): Efficient method for simulation of long-distance gas transport networks with large amounts of hydrogen injection. In: Energy Conversion and Management 234, S. 113984. DOI: 10.1016/j.enconman.2021.113984.
